presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -240,6 +240,9 @@ public HttpRemoteTask(Session session,
                 if (state.isDone()) {
                     cleanUpTask();
                 }
+                else {
+                    partitionedSplitCountTracker.setPartitionedSplitCount(getPartitionedSplitCount());
```

Comment:
Oh I see. I was confused because I thought this state change listener only fired on task state changes, but it actually fires on all task status changes